core/modules/comment/comment.module

```diff
@@ -67,6 +67,22 @@ define('COMMENT_NODE_CLOSED', 1);
  */
 define('COMMENT_NODE_OPEN', 2);
 
+/**
+ * Auto generated comment titles.
+ */
+define('COMMENT_TITLE_AUTO', 'auto');
+
+/**
+ * Custom comment titles.
+ */
+define('COMMENT_TITLE_CUSTOM', 'custom');
+
+/**
+ * Hidden comment titles.
+ */
+define('COMMENT_TITLE_HIDDEN', 'hidden');
+
```

Comment:
Thinking this over... as we now switched to human readable values, there's actually no need for constants anymore.

Constants are usually only used if the value is something obscure, like 1, 4, 123, ...

Reply:
Not true. There are a number of places where constants store strings in core. It's useful for collecting them at the top of their file.

E.g.: `define('MAINTENANCE_MODE', 'install');`

Reply:
Not to mention that the values get referenced all over the place so using the constant is better practice rather than a string directly. It signals that the value is defined somewhere else and not just locally.

Reply:
I think using constants is a good practice, in theory that may reduce typo potential and gives us a place for documentation. However, I found in the .install file we can't (or shouldn't) use constants because the .module file might not always be loaded. So in the update hook we end up using the string values anyway.